narwhals/_pandas_like/dataframe.py

```diff
@@ -370,6 +372,10 @@ def drop_nulls(self, subset: str | list[str] | None) -> Self:
         plx = self.__narwhals_namespace__()
         return self.filter(~plx.any_horizontal(plx.col(*subset).is_null()))
 
+    def estimated_size(self, unit: SizeUnit = "b") -> int | float:
```

Comment:
Similarly as per arrow:

```suggestion
    def estimated_size(self, unit: SizeUnit) -> int | float:
```